### PR TITLE
Added search feature

### PR DESCRIPTION
**Pull request by Shalev-Lifshitz**
Tests still need to be added! Make a PR now (before tests are added) so that we can demo this asap.

### CODE REVIEW

**Review by tupi2001 — COMMENTED** [review]
Parameter names for searchCars method are a bit hard to understand.
```
downpayment
'''
does not follow correct java naming conventions

**Comment by sammdu** [comment]
Also, wee need pagination for search. We may need to even add a new DB table for a user's search results to enable pagination.

**Comment by sammdu** [comment]
image
Here you are always trying to get the result, rather than checking if the user exists.
You should return `null` if there are no results for this query.

Also, before we complete `upstream/BankApi` and add a `/bank` endpoint, Nada's `SvcUserLogin` will not be functional, and we won't have a mechanism for adding new users, so your search will always fail because the user doesn't exist.

Additionally, since we're sticking with Option B, you should have an if statement in `SvcSearch` that returns all cars in the cars table if `userId` is `""` or `null`.

In other words, for this GET request: 
```
http://localhost:8080/search?user_id=&down_payment=5000&budget_mo=250&sort_by=apr&sort_asc=true&keywords=
```
You should check and see that `userId` is `""` (it could also be `null` let's make it more robust by supporting both), disregard all other parameters, and just return all cars in our database.

Later on as we implement search and sort, the `/search` should probably also support keywords and sort order features for all the cars, but for now just return all cars in default order.

**Comment by sammdu** [comment]
- BigDecimal cannot be directly cast to double, you must use the doubleValue() method
- use executeUpdate() for SQL statements that don't expect a return value
- always close the statement before you return

Now the first 4 tests pass, the 5th test doesn't terminate.

**Comment by sammdu** [comment]
Currently, when an invalid user is supplied, it results in an HTTP 500 error:
```
2021-12-01 03:53:37.740 ERROR 6281 --- [nio-8080-exec-7] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0] with root cause

java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
        at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64) ~[na:na]
        at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70) ~[na:na]
        at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248) ~[na:na]
        at java.base/java.util.Objects.checkIndex(Objects.java:372) ~[na:na]
        at java.base/java.util.ArrayList.get(ArrayList.java:459) ~[na:na]
        at tech.autodirect.api.database.TableUsers.getUserByID(TableUsers.java:71) ~[main/:na]
        at tech.autodirect.api.services.SvcSearch.searchCarsWithOffer(SvcSearch.java:131) ~[main/:na]
        at tech.autodirect.api.services.SvcSearch.searchCars(SvcSearch.java:91) ~[main/:na]
        at tech.autodirect.api.ApiEndpoints.search(ApiEndpoints.java:103) ~[main/:na]
```

Instead of forcing a list read, handle an invalid user gracefully by coding the case where the user does not exist explicitly, and return a 404 HTTP error with string `"User not found."`

**Review by sammdu — APPROVED** [review]
Search preliminarily functions
